src/Microsoft.AspNetCore.DataProtection/KeyManagement/KeyBase.cs

```diff
@@ -41,6 +52,24 @@ public IAuthenticatedEncryptorDescriptor Descriptor
             }
         }
 
+        public IAuthenticatedEncryptor CreateEncryptorInstance()
+        {
+            if (_encryptor == null)
+            {
+                foreach (var factory in _encryptorFactories)
+                {
+                    var encryptor = factory.CreateEncryptorInstance(this);
+                    if (encryptor != null)
+                    {
+                        _encryptor = encryptor;
+                        break;
+                    }
+                }
+            }
+
+            return _encryptor;
```

Comment:
Yes and we already do a lock here https://github.com/aspnet/DataProtection/pull/221/files#diff-3dda400bcdb5a80421cdd26389311c63R77. The other places that we call this are just for `Debug.Assert` and we don't use those objects that are created. This has been the same since before the refactor.